### ISSUE TITLE
Simplify different root cases if possible

### ISSUE THREAD

**Issue by tm-ms** [microstream-one/microstream#52]
Internal confusion and ticket \#J117 showed that the two concepts for handling the two different kinds of root instances are very confusing for new users.
It would be a good idea to think about fruther streamlining options for handling roots sooner rather than later.

**Comment by tm-ms** [comment]
Internal root handling has been overhauled with an even bigger scope to handle the two different kinds of root instance (explicit instance passed to initialization or root intances implicitely set after initialization) in one unified way.
This makes the two kinds interchangeable, meaning they are not even distinctable anymore from an API user perspective.
More specifically, the two kinds of roots are interchangeable from one storage start to another.

This creates 4 possible cases. All of which have been tested successfully:

"straight forward" cases:
1.) store explicit root instance (via #start), shutdown, startup again, explicit root gets updated by initialization.
2.) store implicit root instance (via #setRoot after #start), shutdown, startup again, query root as implicit kind (#getRoot)

"switching" cases:
3.) store explicit root instance (via #start), shutdown, remove explicit instance, startup again, query root as implicit kind (#getRoot)
4.) store implicit root instance (via #setRoot after #start), shutdown, add explicit instance, startup again, explicit root gets updated by initialization.

Of course, the change must be backwards-compatible to existing persisted old roots ("customtRoot", "defaultRoot").

This creates another 4 possible cases., which have also been tested successfully:

1.) explicit root instance (via #start), customRoot exists in the database. This associated and updates the explicit root instance with the custom root instance. Of course the object's type must fit.

2.) no root instance exists, customRoot exists in the database. This loads and sets the customRoot as an implicit root instance.

3.) explicit root instance (via #start), defaultRoot exists in the database. This associated and updates the explicit root instance with the root instance referenced by the defaultRoot instance. Of course the object's type must fit.

4.) no root instance exists, defaultRoot exists in the database. This loads and sets the root instance referenced by the defaultRoot instance as an implicit root instance.


All four cases store the changed Storage roots instance so that after the first start with the new version, the new root handling is persisted and the old one is gone (more precisely unreachable and subject to storage garbage collection).


**In short**:
There seemingly now is only "one" unified type of root handling including backwards compatibility.
Anything about root handling can be changed at any time and the root instance will always be loaded accordingly (provided an explicit root instance's class fits to the persisted data, of course).

I hope this removes all the enormous confusion that two different kinds of root handling strategies are two different things.
